4_Control panel/Python_GUI_day4.py

```diff
@@ -1,123 +1,315 @@
+        dpg.add_checkbox(label="Intracranial Pressure", pos=[10,225],default_value=True, callback=lambda sender, data: dpg.configure_item("y_axis3", show=data))
 
-    dpg.create_viewport(title='Multiple channel display', width=800, height=700)
-    dpg.setup_dearpygui()
-    dpg.show_viewport()
-    i = 0
+        # The fowllowing codes are used to create a color picker to change the color of the plot
+        dpg.add_text("Color selection:",pos=[10,260])
+        dpg.add_color_picker(label="Color", default_value=[255, 0, 0, 255], pos=[10,285], width=200, tag="color_picker")
+        dpg.add_text("Set color for:",pos=[10,545])
+        dpg.add_combo(items=["ABP", "ECG", "ICP"], default_value="ABP", pos=[110,545], width=50, tag="color_combo")
+        dpg.add_button(label="SET", callback=change_color_func, pos=[170,540], width=30, height=30)
+        # To change the color of item in dearpygui, we need to create a theme with specified color and bind the theme to the item
+        # Create three themes for three channels
+        global colors_name
+        colors_name = ["ABP_color", "ECG_color", "ICP_color"] 
+        tags_theme = "ABP_theme", "ECG_theme", "ICP_theme"
+        for i,(tag_theme, tag) in enumerate(zip(tags_theme, TAGS)):
+            with dpg.theme(tag=tag_theme):
+                with dpg.theme_component(dpg.mvLineSeries):
+                    globals()[colors_name[i]] = dpg.add_theme_color(dpg.mvPlotCol_Line, (51, 255, 255), category=dpg.mvThemeCat_Plots)
+            # bind the theme to the item, when we change the color the the theme, the color of the item will be changed
+            dpg.bind_item_theme(tag, tag_theme)
+
+        # The following codes are used to create a text to show the analysis result
+        dpg.add_text("Analysis result:",pos=[10,580])
+        dpg.add_text("Heart rate:",pos=[10,605])
+        # The heart rate will be calculated and updated
+        dpg.add_text("60bpm",tag="text_heart_rate",pos=[100,605])
+
+def create_file_selection_dialog():
+    """ create a file selection dialog
+    This function is used to create a file selection dialog
+    This dialog is used to select the data file
     """
+    with dpg.file_dialog(directory_selector=False, show=False,callback=file_selection_func, tag="file_selection_dialog", width=500, height=500):
+        dpg.add_file_extension(".dat")
+
+def start_dearpygui(ABP,ECG,ICP):
+    """ update the GUI frame by frame
     Here we change dpg.start_dearpygui() to while dpg.is_dearpygui_running() and dpg.render_dearpygui_frame()
     This is because we want to update the plot widget in a loop, if we use dpg.start_dearpygui(), the GUI will
     be frozen and the plot widget will not be updated.
-    """
+
+    Args:
+        ABP: the data of ABP
+        ECG: the data of ECG
+        ICP: the data of ICP
+
+    """  
+    i = 0
     while dpg.is_dearpygui_running():
+        if i >= len(ICP) or start_stop == 0:
+            i = 0
         i = i + 1
```

Comment:
If you implement my suggestion above, you need to skip this if in pause.
```suggestion
        if not pause: 
            i = i + 1
```